mapper/namespace_reference.go

```diff
@@ -32,13 +32,10 @@ func (n *NamespaceReference) FromInternal(data map[string]interface{}) {
 
 func (n *NamespaceReference) ToInternal(data map[string]interface{}) error {
 	namespaceID, ok := data["namespaceId"]
-	if !ok {
-		return nil
-	}
 	for _, path := range n.fields {
 		convert.Transform(data, path, func(input interface{}) interface{} {
 			parts := strings.SplitN(convert.ToString(input), ":", 2)
-			if len(parts) == 2 && (parts[0] == namespaceID) {
+			if len(parts) == 2 && (!ok || parts[0] == namespaceID) {
```

Comment:
@gitlawr this change doesn't seem to cover the case when namespaceID is missing and len(parts) != 2 - previously namespaceId check was covering this situation. Should we be concerned?